users/user-guide-query/multi-stage-query/operator-types/hash_join.md

```diff
@@ -0,0 +1,210 @@
+
+The [join_overflow_mode](../hints/join_overflow_mode.md) hint can be used to control the behavior of the engine when the
+number of groups exceeds the limit.
+This limit can be defined using the [max_rows_in_join](../hints/max_rows_in_join.md) hint.
+By default, this limit is slightly above 1 million groups and the default join overflow mode is `THROW`,
+which means that the query will fail if the number of groups exceeds the limit.
+
+## Hints
+
+### join_overflow_mode
+Type: String
+
+Default: `THROW`
+
+Defines the behavior of the engine when the number of groups exceeds the limit defined by the `max_rows_in_join` hint.
+The possible values are:
+- `THROW`: The query will fail if the number of groups exceeds the limit.
+- `BREAK`: The engine will stop processing the join and return the results that have been computed so far. In this case
+  the stat `maxRowsInJoinReached` will be true.
+
+### max_rows_in_join
+Type: Integer
+
+Default: 1.048.576
+
+The maximum number of groups that can be created by the join keys. What happens when this limit is reached is defined by
+the `join_overflow_mode` hint.
+
+{% hint style="warn" %}
+Take care when increasing this limit. If the number of groups is too high, the amount of memory used by the engine can
+be very high, which can lead to very large GC pauses and even out of memory errors.
+{% endhint %}
+
+<!-- TODO
+### join_strategy
+Type: List of Strings separated by commas.
+
+Default: `""`
+If set, the engine will use the join strategy defined in the hint. The possible values are:
+- `hash_table`: The default join strategy. The engine will use a hash join algorithm to join the relations.
+- `dynamic_broadcast`: The engine will use a broadcast join algorithm to join the relations. This is useful when the
+  right relation is small enough to be broadcasted to all workers.
+
+When this hint is not defined (or if it is defined to `""`), the engine will use the hash join algorithm in most of the
+cases but will automatically try to use the broadcast join algorithm used to optimize SEMI-JOIN queries.
+
+{% hint style="warn" %}
+By providing this hint, the engine will try to use the join strategy defined in the hint, even if it is not the most 
+efficient. This can be useful to test the performance of different join strategies in the SEMI-JOIN case or to
+disable co-located joins.
+It is not recommended to use this hint in normal queries.
+{% endhint %}
+
+### is_colocated_by_join_keys
+END OF TODO -->
+
+## Stats
+### executionTimeMs
+Type: Long
+
+The summation of time spent by all threads executing the operator.
+This means that the wall time spent in the operation may be smaller that this value if the parallelism is larger than 1.
+
+### emittedRows
+Type: Long
+
+The number of groups emitted by the operator.
+Joins can emit more rows than the input relations, so this value can be higher than the number of rows in the input.
```

Comment:
Joins can emit more rows than the input relations - how is this possible?

Reply:
Joins can emit `O(r^l)` rows. That is easy to see in a cross join but even in a inner join. For example in postgres:

```
postgres=# insert into two_cols (a, b) values (1, 1), (1, 2), (10, 10);
INSERT 0 3
postgres=# select * from two_cols;
 a  | b  
----+----
  1 |  1
  1 |  2
 10 | 10
(3 rows)

postgres=# select * from two_cols as a1 join two_cols as a2 on a1.a = a2.a;
 a  | b  | a  | b  
----+----+----+----
  1 |  1 |  1 |  1
  1 |  1 |  1 |  2
  1 |  2 |  1 |  1
  1 |  2 |  1 |  2
 10 | 10 | 10 | 10
(5 rows)
```

Reply:
oh yes of course (⸝⸝⸝O﹏ O⸝⸝⸝) joins 101..